### DISCUSSION TITLE
[Proposal] Force-Resolver Directive

### DISCUSSION THREAD

**Discussion by Panakotta00** [99designs/gqlgen#2166] [Ideas]
I find it a bit frustrating when working on nested models that you have to constantly change the graphql config
to force the usage of a resolver for a given field.
In my case most fields have to be resolved anyway.

Force Resolve Opt-In is not the matter,
what I find annyoing is that i have to change the config to enforce the usage of a resolver.

To me this information should be in the schema file.

For this i propose a "forceResolver" directive.

Imagine following schmea:

```graphql 
type Inner {
    data1: String!
    data2: String!
}

type Outer {
    data1: String!
    data2: String!
    inner: Inner!
}
```

For the `inner` field of the `Outer` type to be resolved using a resolver i would have to add
```yml
models:
  Outer:
    fields:
      inner:
        resolver: true
```

My feature request would ba `forceResolver` directive:
```graphql 
type Inner {
    data1: String!
    data2: String!
}

type Outer {
    data1: String!
    data2: String!
    inner: Inner! @forceResolver
}
```

That removes the necessety to add information of the schematic to the config, which should be in my opinion sepperated.

**Comment by lleadbet** [comment]
This already exists as a directive- `@goField(forceResolver:true)`; https://gqlgen.com/config/#inline-config-with-directives
